js/src/setup-mc/setup-stepper/index.js

```diff
@@ -9,29 +9,60 @@ import { getHistory, getNewPath } from '@woocommerce/navigation';
 import AppSpinner from '.~/components/app-spinner';
 import SavedSetupStepper from './saved-setup-stepper';
 import useMCSetup from '.~/hooks/useMCSetup';
+import useStoreAddress from '.~/hooks/useStoreAddress';
+import useGoogleMCPhoneNumber from '.~/hooks/useGoogleMCPhoneNumber';
 import stepNameKeyMap from './stepNameKeyMap';
 
 const SetupStepper = () => {
 	const { hasFinishedResolution, data: mcSetup } = useMCSetup();
+	const { data: address, loaded: addressLoaded } = useStoreAddress();
+	const { data: phone, loaded: phoneLoaded } = useGoogleMCPhoneNumber();
 
-	if ( ! hasFinishedResolution && ! mcSetup ) {
-		return <AppSpinner />;
-	}
+	const hasValidPhoneNumber =
+		phoneLoaded && phone?.isValid && phone?.isVerified;
+
+	const hasValidAddress =
+		addressLoaded &&
+		address?.isAddressFilled &&
+		! address?.isMCAddressDifferent;
+
+	const hasConfirmedStoreRequirements =
+		hasValidPhoneNumber && hasValidAddress;
+
+	const hasLoaded =
+		hasFinishedResolution && mcSetup && addressLoaded && phoneLoaded;
 
 	if ( hasFinishedResolution && ! mcSetup ) {
 		// this means error occurred, we just need to return null here,
 		// wp-data actions will display an error snackbar at the bottom of the page.
 		return null;
 	}
 
-	const { status, step } = mcSetup;
+	if ( ! hasLoaded ) {
+		return <AppSpinner />;
+	}
+
+	const { status } = mcSetup;
+	let { step } = mcSetup;
+
+	// If the user has already completed the store requirements, but is currently still on the
+	// store requirements step, we should skip the store requirements step and go to the paid ads step.
+	// else they will get stuck on a non-existent step #3
+	if ( step === 'store_requirements' && hasConfirmedStoreRequirements ) {
+		step = 'paid_ads';
+	}
```

Comment:
Smart. Rather than mutating the `step` variable here, it may be better to do something like this:

```js
const { status, step } = mcSetup;

const currentStep = ( step === 'store_requirements' && hasConfirmedStoreRequirements ) ?  'paid_ads' : step;
```